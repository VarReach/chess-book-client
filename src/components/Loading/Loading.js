import React, { Component } from 'react';
//courtesy of @tobiasahiin (tobiasahlin.com/spinkit)
import './Loading.css';

let timeout;

export default class Loading extends Component {
  state = {
    timein: null,
  }

  componentDidMount() {
    timeout = setTimeout(() => this.setState({ timein: true }), 1000); //has to be loading for at least a second to show the loading animation
  }

  componentWillUnmount() {
    clearTimeout(timeout);
  }

  render() {
    if (this.props.status && this.state.timein) {
      return (
        <div className="error-message">
          <h1>Uh oh!</h1>
          <p>Something has gone wrong on our end.</p>
          <p>Try refreshing the page in a bit.</p>
        </div>
      );
    }
    if (this.state.timein) {
      return (
        <div className="spinner">
          <div className="rect1"></div>
          <div className="rect2"></div>
          <div className="rect3"></div>
          <div className="rect4"></div>
          <div className="rect5"></div>
        </div>
      );
    }
    return null;
  }
}