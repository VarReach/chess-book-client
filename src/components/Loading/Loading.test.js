import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Loading from './Loading';

describe('Loading component', () => {
  it('renders the component', () => {
    const wrapper = shallow(<Loading/>)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the loading animation after 1 second', (done) => {
    const wrapper = shallow(<Loading/>);
    setTimeout(() => {
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    }, 1200);
  });

  it('render the error view when an error is in props', () => {
    const wrapper = shallow(<Loading status={true}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});