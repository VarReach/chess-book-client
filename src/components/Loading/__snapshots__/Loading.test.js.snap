// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component render the error view when an error is in props 1`] = `""`;

exports[`Loading component renders the component 1`] = `""`;

exports[`Loading component renders the loading animation after 1 second 1`] = `
<div
  className="spinner"
>
  <div
    className="rect1"
  />
  <div
    className="rect2"
  />
  <div
    className="rect3"
  />
  <div
    className="rect4"
  />
  <div
    className="rect5"
  />
</div>
`;
